Python/bltinmodule.c

```diff
@@ -347,6 +347,53 @@ builtin_all(PyObject *module, PyObject *iterable)
     Py_RETURN_TRUE;
 }
 
+static PyObject *
+builtin_all_equal(PyObject *module, PyObject *iterable)
+{
+    PyObject *it, *item, *prev_item;
+    PyObject *(*iternext)(PyObject *);
+    int cmp;
+
+    it = PyObject_GetIter(iterable);
+    if (it == NULL)
+        return NULL;
+    iternext = *Py_TYPE(it)->tp_iternext;
+
+    prev_item = iternext(it);
+    if (prev_item == NULL) {
+        Py_RETURN_TRUE;
+    }
+
+    for (;;) {
+        item = iternext(it);
```

Comment:
Why not put it in a `while` loop? Like this:
```c
while(item = iternext(it))
```

Reply:
Could do I suppose. I doubt it would be well received if I were to submit a patch to change the existing functions, but that's a separate issue. I mostly built from what was already there. In some ways, I suppose this is more descriptive than a "for infinity" loop.